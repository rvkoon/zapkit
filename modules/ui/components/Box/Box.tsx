import { ReactNode } from "react";
import cx from "classnames";

export interface IBoxProps {
  children: ReactNode;
  className?: string;
  isStickingSides?: boolean;
}

export function Box({
  children,
  className,
  isStickingSides = false,
}: IBoxProps): JSX.Element {
  return (
    <div className={cx("px-1", isStickingSides && "px-0")}>
      <div className={cx("rounded-base p-base bg-white", className)}>
        {children}
      </div>
    </div>
  );
}
