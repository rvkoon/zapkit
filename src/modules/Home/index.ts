export * from './components';
export * from './helpers';
export * from './hooks';
export * from './services';
export * from './types';
export * from './constants';