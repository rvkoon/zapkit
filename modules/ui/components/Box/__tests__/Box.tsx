import { Box, IBoxProps } from "../Box";
import { RenderResult } from "@testing-library/react";
import { render } from "modules/tests";

describe("Box component", () => {
  function renderComponent(props: IBoxProps): RenderResult {
    return render(<Box {...props} />);
  }

  it("should render correctly", () => {
    const { container } = renderComponent({ children: <div></div> });
    expect(container).toMatchSnapshot();
  });
});
