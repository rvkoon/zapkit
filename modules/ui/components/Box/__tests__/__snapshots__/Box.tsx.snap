// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component should render correctly 1`] = `
<div>
  <div
    class="px-1"
  >
    <div
      class="rounded-base p-base bg-white"
    >
      <div />
    </div>
  </div>
</div>
`;
